core/slice.go

```diff
@@ -601,12 +601,26 @@ func (sl *Slice) writeToPhCache(pendingHeaderWithTermini types.PendingHeader) {
 
 // pickPhCacheHead determines if the provided pendingHeader should be selected and returns true if selected
 func (sl *Slice) pickPhCacheHead(reorg bool, externPendingHeaderWithTermini types.PendingHeader, domOrigin bool) bool {
+	localPendingHeader, exists := sl.phCache[sl.pendingHeaderHeadHash]
+
 	if reorg {
-		sl.pendingHeaderHeadHash = externPendingHeaderWithTermini.Termini[terminiIndex]
-		return true
-	}
+		// Knot exscape
+		if !exists {
```

Comment:
A) this escape is too loose -- sounds like it is only needed for block no 1?
B) do we need escapes at all? I would prefer it `Slice.init()` just wrote an empty ph to the cache for block 1, or something like that